File: php/src/Model/WorkerBreak.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * One break entry, open or closed.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerBreak
{
    public function __construct(
        public readonly string $id,
        public readonly string $startedAt,
        /** null while the break is open */
        public readonly ?string $endedAt,
        public readonly ?string $reason,
        public readonly int $durationMs,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            startedAt: (string) ($data['startedAt'] ?? ''),
            endedAt: isset($data['endedAt']) ? (string) $data['endedAt'] : null,
            reason: isset($data['reason']) ? (string) $data['reason'] : null,
            durationMs: (int) ($data['durationMs'] ?? 0),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/WorkerBreak.java
package io.fivexer.sdk.model;

/**
 * One break entry, open or closed.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerBreak {
    private String id;
    private String startedAt;
    private String endedAt;  // null while the break is open
    private String reason;
    private long durationMs;

    public String getId() { return id; }
    public String getStartedAt() { return startedAt; }
    public String getEndedAt() { return endedAt; }
    public String getReason() { return reason; }
    public long getDurationMs() { return durationMs; }
}
